CMakeLists.txt

```diff
@@ -32,18 +26,48 @@ else()
   message(FATAL_ERROR "Required dependency GMP not found. Set GMP_LIB_DIR and GMP_INCLUDE_DIR to inform cmake where to look")
 endif()
 
+# Find PYTHON_EXECTUABLE first based on $PATH
+find_package(PythonInterp)
 if (PYTHONINTERP_FOUND)
-  message(STATUS "Python binary found at ${PYTHON_EXECUTABLE}. Building python bindings")
+  message(STATUS "Python binary found at ${PYTHON_EXECUTABLE}")
+
   execute_process(
     COMMAND
       ${PYTHON_EXECUTABLE} -c "import numpy, sys;sys.stdout.write(numpy.get_include())"
     OUTPUT_VARIABLE NUMPY_INCLUDE_DIRS
   )
+
+  # This uses sysconfig from inside PYTHON_EXECTUABLE to ensure we link against same python executable that is present in $PATH
+  execute_process(
+    COMMAND
+      ${PYTHON_EXECUTABLE} -c "import sysconfig, sys;sys.stdout.write(sysconfig.get_config_var('LIBPC'))"
+    OUTPUT_VARIABLE PYTHON_LIBPC
+  )
+
   message(STATUS "Numpy found in ${NUMPY_INCLUDE_DIRS}")
-  message(STATUS "Using python libraries in ${PYTHON_LIBRARIES}")
-  message(STATUS "Using python headers in ${PYTHON_INCLUDE_DIRS}")
+  message(STATUS "Looking for python pkgconfig in ${PYTHON_LIBPC}")
+
+  set(ENV{PKG_CONFIG_PATH} "${PYTHON_LIBPC}:$ENV{PKG_CONFIG_PATH}")
+  pkg_check_modules(PYTHON python)
+
+  # PYTHON_LIBRARIES and PYTHON_LIBRARY_DIRS have the name and path to where python library exists
+  # We need to combine those and add appropriate prefixes and suffixes so we can pass library directly later
+  # Use a name other than PYTHON_LIBRARY since find_library won't do anything if variable is already set
+  unset(_PYTHON_LIBRARY)
+  find_library(_PYTHON_LIBRARY NAMES ${PYTHON_LIBRARIES} PATHS ${PYTHON_LIBRARY_DIRS} NO_DEFAULT_PATH)
+  if(_PYTHON_LIBRARY)
+    set(PYTHON_LIBRARY ${_PYTHON_LIBRARY})
+    message(STATUS "Python library found at ${PYTHON_LIBRARY}")
+  else()
+    message(WARNING " Unable to find full path to python lib")
+    set(PYTHON_FOUND FALSE)
+  endif()
 else()
-  message(STATUS "Python not found. Python bindings will not be built.")
+  message(WARNING "No python interpreter found!")
+endif()
+
+if(NOT PYTHON_FOUND)
+  message(STATUS "Python libs not found. Python bindings will not be built.")
```

Comment:
This seems to be being hit in jenkins. Is that a jenkins configuration bug with missing libraries?